test/functional/vimscript/api_functions_spec.lua

```diff
@@ -48,10 +49,46 @@ describe('eval-API', function()
     eq('Vim(call):E5555: API call: Invalid buffer id: 17', err)
   end)
 
-  it('cannot change texts if textlocked', function()
+  it('cannot change text or window if textlocked', function()
     command("autocmd TextYankPost <buffer> ++once call nvim_buf_set_lines(0, 0, -1, v:false, [])")
     matches('Vim%(call%):E5555: API call: E565: Not allowed to change text or change window$',
        pcall_err(command, "normal! yy"))
+
+    command("autocmd TextYankPost <buffer> ++once call nvim_open_term(0, {})")
+    matches('Vim%(call%):E5555: API call: E565: Not allowed to change text or change window$',
+       pcall_err(command, "normal! yy"))
+
+    -- Functions checking textlock should also not be usable from <expr> mappings.
+    command("nnoremap <expr> <f2> nvim_win_close(0, 1)")
+    eq('Vim(normal):E5555: API call: E565: Not allowed to change text or change window',
+       pcall_err(command, [[execute "normal \<f2>"]]))
```

Comment:
```suggestion
    command("inoremap <expr> <f2> nvim_win_close(0, 1)")
    eq('Vim(normal):E5555: API call: E565: Not allowed to change text or change window',
       pcall_err(command, [[execute "normal i\<f2>"]]))
```

Reply:
Good ideas, thanks! I'll apply them in a jiffy. :+1: